src/plugins/data/common/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -100,6 +102,7 @@ export class IndexPattern implements IIndexPattern {
   public sourceFilters?: SourceFilter[];
   public dataSourceRef?: SavedObjectReference;
   public fieldsLoading?: boolean;
+  public signalType?: SignalType;
```

Comment:
this should be just a string.

`signalType` doesn't isn't limited to observability but based on this enum it is. it's fine that we have predefined enums but this field shouldn't be reserved for observability datasets only